data/basics/address.go

```diff
@@ -77,9 +80,11 @@ func UnmarshalChecksumAddress(address string) (Address, error) {
 
 // String returns a string representation of Address
 func (addr Address) String() string {
-	var addrWithChecksum []byte
-	addrWithChecksum = append(addr[:], addr.GetChecksum()...)
-	return base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(addrWithChecksum)
+	addrWithChecksum := make([]byte, crypto.DigestSize+checksumLength)
+	copy(addrWithChecksum[:crypto.DigestSize], addr[:])
+	shortAddressHash := crypto.Hash(addr[:])
```

Comment:
why not using addr.GetChecksum() ? It would make the code more readable and seems like there is almost no performance penalty on this extra call